src/vm/exceptionhandling.cpp

```diff
@@ -3670,7 +3640,7 @@ ExceptionTracker* ExceptionTracker::GetOrCreateTracker(
             fIsRethrow = true;
         }
         else
-        if ((pTracker->m_ptrs.ExceptionRecord != pExceptionRecord) && !fIsInterleavedHandling)
+        if ((pTracker->m_ptrs.ExceptionRecord != pExceptionRecord) && fIsFirstPass)
```

Comment:
After we catch the PAL_SEHException in the DispatchManagedException, we need to create a new instance of the exception and let the catch handler complete to destroy the originally thrown one to prevent a memory leak in the native exception handling code. The same thing happens when we unwind a sequence of managed frames by throwing the exception. Again, when we catch it, we need to let the catch handler complete and destroy the thrown exception and so we create a new instance that we pass to the UnwindManagedExceptionPassX.
That means that the original condition (I mean just the exception record comparison) would indicate that it is a new exception, since the exception records are different from the ones the exception tracker has seen before.
Now this happens only during second pass of the exception handling. And it is not possible to get a new exception that is also in second pass at that point. So the comparison of exception records make sense only if the current exception handling is in first pass. And that's what the condition checks.
I believe this added condition should be fine on Windows, as the same is true there - we can only get a different exception when it is in the first pass of its processing.
Does it make sense?